### PR TITLE
Fixes #8671: Generate the json file to be able to create a directive in the technique testing scenario

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8671

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

File: rtf [hunk 1]
```diff
@@ -21,6 +21,7 @@ Usage:
     rtf scenario list
     rtf scenario env <platform>
     rtf scenario run <platform> <scenario> [--no-finally] [--stop] [--filter=<test1>,<test2>,...] [--format=<format>]
+    rtf scenario technique from-directive <platform> <uuid> <test_name>
```
Comment (amousset) [inline rtf]: test_name is not used, I will remove it.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
@peckpeck Ready for review.

File: README.md [hunk 2]
```diff
@@ -86,6 +86,20 @@ A typical scenario:
...
 
+Writing a technique test
+------------------------
+
+Technique tests use a special scenario to test a technique. The are localed in the technique folder,
+in a `tests` directory, containing:
+
+- `my_test.metadata`: The main file, in JSON, containing information about the test, for example the global compliance and the paths of other test files.
+- `my_test.json`: The directive parameters for the test.
+- `my_test.rb`: The serverspec file containing the actual test. It should check if the directive was corrently applied.
```
Comment (peckpeck) [inline README.md]: corrently ?

File: rtf [hunk 3]
```diff
@@ -763,6 +769,43 @@ Add the test to an existing scenario:
...
+    (rudder_url, token) = self.api_connection_info()
+    endpoint = RudderEndPoint(rudder_url, token, verify=False)
+
+    file_check = test_name + ".rb"
+    file_metadata = test_name + ".metadata"
+    file_directive = test_name + ".json"
+
+    # Generate metadata
+    metadata = {}
+    metadata["check"] = file_metadata
```
Comment (peckpeck) [inline rtf]: this should be file_check

**Comment by amousset** [comment]
Commit modified

File: rtf [hunk 4]
```diff
@@ -763,6 +769,43 @@ Add the test to an existing scenario:
...
+    # Generate check file
+    make_user_testfile(os.path.join(path, file_check))
+
+    # Generate directive
+    directive = endpoint.directive_details(directive_uuid)['directives'][0]
+
+    # This is needed because of #8687, the API does not know these parameters at creation
+    del directive["isEnabled"]
+    del directive["isSystem"]
+    del directive["priority"]
```
Comment (peckpeck) [inline rtf]: Are they always present ? shouldn't there be a test around them ?

File: rtf [hunk 5]
```diff
@@ -763,6 +769,43 @@ Add the test to an existing scenario:
...
+
+    make_jsonfile(os.path.join(path, file_directive), directive)
+
```
Comment (peckpeck) [inline rtf]: We should print a little note here, like in export_test, to tell the user what to do next.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by peckpeck** [comment]
Good, merging

**Pull request merged**